openforcefield/topology/molecule.py

```diff
@@ -1888,64 +1887,185 @@ def from_smiles(smiles,
+            # always match by atleast atomic number
+            is_equal = (x['atomic_number'] == y['atomic_number'])
+            if aromatic_matching:
+                is_equal &= (x['is_aromatic'] == y['is_aromatic'])
+            if formal_charge_matching:
+                is_equal &= (x['formal_charge'] == y['formal_charge'])
+            if atom_stereochemistry_matching:
+                is_equal &= (x['stereochemistry'] == y['stereochemistry'])
             return is_equal
 
-        def edge_match_func(x, y):
-            # We don't need to check the exact bond order (which is 1 or 2)
-            # if the bond is aromatic. This way we avoid missing a match only
-            # if the alternate bond orders 1 and 2 are assigned differently.
-            is_equal = x['is_aromatic'] == y['is_aromatic'] or x['bond_order'] == y['bond_order']
-            if compare_bond_stereochemistry:
-                is_equal &= x['stereochemistry'] == y['stereochemistry']
-            return is_equal
+        # check if we want to do any bond matching if not the function is None
+        if aromatic_matching or bond_order_matching or bond_stereochemistry_matching:
+            def edge_match_func(x, y):
+                # We don't need to check the exact bond order (which is 1 or 2)
+                # if the bond is aromatic. This way we avoid missing a match only
+                # if the alternate bond orders 1 and 2 are assigned differently.
+                if aromatic_matching and bond_order_matching:
+                    is_equal = (x['is_aromatic'] == y['is_aromatic']) or (x['bond_order'] == y['bond_order'])
+                elif aromatic_matching:
+                    is_equal = (x['is_aromatic'] == y['is_aromatic'])
+                elif bond_order_matching:
+                    is_equal = (x['bond_order'] == y['bond_order'])
+                else:
+                    is_equal = None
+                if bond_stereochemistry_matching:
+                    if is_equal is None:
+                        is_equal = (x['stereochemistry'] == y['stereochemistry'])
+                    else:
+                        is_equal &= (x['stereochemistry'] == y['stereochemistry'])
+
+                return is_equal
+        else:
+            edge_match_func = None
+
+        # Here we should work out what data type we have, also deal with lists?
+        def find_data_type(data):
+            """Find the data type and return the networkx graph"""
+
+            try:
+                # Molecule class instance
+                return data.to_networkx()
+            except AttributeError:
+                try:
+                    # TopologyMolecule class instance
+                    return data.reference_molecule.to_networkx()
+                except AttributeError:
+                    if isinstance(data, nx.Graph):
+                        return data
+
+        mol1_netx = find_data_type(mol1)
```

Comment:
It would be helpful if the name of this function is more descriptive, e.g. `to_networkx`.

Reply:
Good idea I think `to_networkx` makes sense here.